tests/lib/rules/no-whitespace-before-property.js

```diff
@@ -507,6 +510,46 @@ ruleTester.run("no-whitespace-before-property", rule, {
             code: "foo ['bar ' + 1 + ' baz']",
             output: "foo['bar ' + 1 + ' baz']",
             errors: ["Unexpected whitespace before property 'bar ' + 1 + ' baz'."]
+        },
+        {
+            code: "5       .toExponential()",
+            output: "5 .toExponential()",
```

Comment:
I think there are 4 options for how to write a property like this:

1. `5 .toExponential()`
1. `5..toExponential()`
1. `5.0.toExponential()`
1. `(5).toExponential()`

If the user has a rule called `no-whitespace-before-property` enabled, they won't want option 1 (since that has whitespace before the property). If people prefer option 3 over option 2, then they can enable `no-floating-decimal`. So I think the only thing that we should be concerned about is the choice between `5..toString()` and `(5).toString()`.

If we do decide to fix this case, I think `5..toString()` would be a better choice, since it changes the code less. However, I agree that we can't be sure which the user would prefer.